nx-X11/programs/Xserver/hw/nxagent/Window.c

```diff
@@ -2674,6 +2686,18 @@ void nxagentDisconnectWindow(void * p0, XID x1, void * p2)
   }
   #endif
 
+  if ((nxagentRealWindowProp) && (!nxagentWindowTopLevel(pWin)))
+  {
+    Atom prop = MakeAtom("NX_REAL_WINDOW", strlen("NX_REAL_WINDOW"), True);
```

Comment:
> Should we cache this atom instead of re-creating it for every window creation and deletion?

No. I played with nxagentMakeAtom() (if that's what you refer to). The caching, as I understood it, is something for mapping "remote" and "local" atoms and caching the mapping.

Not for us, it seems. I don't think the overhead is too expensive.